File: matiasmasso/Mat.NET/DTO/DTODeliveryTraspas.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTODeliveryTraspas : DTOBaseGuid
    {
        public DTOEmp Emp { get; set; }
        public int Id { get; set; }
        public DateTime Fch { get; set; }
        public DTOMgz MgzFrom { get; set; }
        public DTOMgz MgzTo { get; set; }
        public List<DTODeliveryItem> Items { get; set; }
        public DTOUsrLog UsrLog { get; set; }

        public DTODeliveryTraspas() : base()
        {
            Items = new List<DTODeliveryItem>();
        }

        public DTODeliveryTraspas(Guid oGuid) : base(oGuid)
        {
            Items = new List<DTODeliveryItem>();
        }

        public static DTODeliveryTraspas Factory(DTOUser oUser, DTOMgz oMgzFrom, DTOMgz oMgzTo, DateTime DtFch = default(DateTime))
        {
            if (DtFch == default(DateTime))
                DtFch = DTO.GlobalVariables.Today();
            DTODeliveryTraspas retval = new DTODeliveryTraspas();
            {
                var withBlock = retval;
                withBlock.Emp = oUser.Emp;
                withBlock.MgzFrom = oMgzFrom;
                withBlock.MgzTo = oMgzTo;
                withBlock.Fch = DtFch;
                withBlock.UsrLog = DTOUsrLog.Factory(oUser);
            }
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTODeliveryTraspas.vb
Public Class DTODeliveryTraspas
    Inherits DTOBaseGuid
    Property Emp As DTOEmp
    Property Id As Integer
    Property Fch As Date
    Property MgzFrom As DTOMgz
    Property MgzTo As DTOMgz
    Property Items As List(Of DTODeliveryItem)
    Property UsrLog As DTOUsrLog

    Public Sub New()
        MyBase.New
        _Items = New List(Of DTODeliveryItem)
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        _Items = New List(Of DTODeliveryItem)
    End Sub

    Shared Function Factory(oUser As DTOUser, oMgzFrom As DTOMgz, oMgzTo As DTOMgz, Optional DtFch As Date = Nothing) As DTODeliveryTraspas
        If DtFch = Nothing Then DtFch = DateTime.Today
        Dim retval As New DTODeliveryTraspas
        With retval
            .Emp = oUser.Emp
            .MgzFrom = oMgzFrom
            .MgzTo = oMgzTo
            .Fch = DtFch
            .UsrLog = DTOUsrLog.Factory(oUser)
        End With
        Return retval
    End Function
End Class
